src/app/space/create/deployments/apps/deployment-details.component.spec.ts

```diff
@@ -152,4 +159,38 @@ describe('DeploymentDetailsComponent', () => {
       expect(de.componentInstance.valueUpperBound).toEqual(4);
     });
   });
+
+  describe('sparkline data', () => {
+    it('should by default be greater or equal to 1', function (this: Context) {
+      let detailsComponent = this.testedDirective;
+      expect(detailsComponent.getSparklineMaxElements()).toBeGreaterThan(0);
+    });
+
+    it('should not be able to be set to anything less than 1', function (this: Context) {
+      let detailsComponent = this.testedDirective;
+      [0, -5, -1873].forEach(n => {
+        detailsComponent.setSparklineMaxElements(n);
+        expect(detailsComponent.getSparklineMaxElements()).toBe(1);
+      })
+    });
+
+    it('should have its cpu data bounded when enough data has been emitted', function (this: Context) {
+      const MAX_CPU_SPARKLINE_ELEMENTS = 4;
+      let detailsComponent = this.testedDirective;
+      detailsComponent.setSparklineMaxElements(MAX_CPU_SPARKLINE_ELEMENTS);
+      times(MAX_CPU_SPARKLINE_ELEMENTS, () => cpuStatObservable.next({ used: 1, quota: 2 }));
+      expect(detailsComponent.cpuData.xData.length).toBe(MAX_CPU_SPARKLINE_ELEMENTS + 1);
+      expect(detailsComponent.cpuData.yData.length).toBe(MAX_CPU_SPARKLINE_ELEMENTS + 1);
+    });
+
+    it('should have its memory data bounded when enough data has been emitted', function (this: Context) {
+      const MAX_MEM_SPARKLINE_ELEMENTS = 6;
+      let detailsComponent = this.testedDirective;
+      detailsComponent.setSparklineMaxElements(MAX_MEM_SPARKLINE_ELEMENTS);
+      times(MAX_MEM_SPARKLINE_ELEMENTS, () => memStatObservable.next({ used: 3, quota: 4, units: 'GB' }));
```

Comment:
Shouldn't these attempt to input more than the MAX times? e.g MAX_MEM_SPARKLINE_ELEMENTS + 10 or something?